### PR TITLE
bpf, xdp: migrate prefilter program to generic __ctx_buff

### PR DESCRIPTION
**Pull request by borkmann**
Few more intermediate cleanups to get out of the pipe in meantime. Most notably to move the prefilter code over to generic __ctx_buff we can later build on top of.

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/10404)

### CODE REVIEW

**Comment by borkmann** [comment]
test-me-please

File: bpf/include/bpf/verifier.h [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+#ifndef __BPF_VERIFIER__
+#define __BPF_VERIFIER__
+
+/**
+ * relax_verifier is a dummy helper call to introduce a pruning checkpoing
```
Comment (qmonnet) [inline bpf/include/bpf/verifier.h]: Nit: can we take this chance to fix the typo on “checkpoing” (and possibly the unnecessary double-star on the line above)?

**Review by qmonnet — APPROVED** [review]
Looks good!

File: bpf/include/bpf/verifier.h (hunk 1 above)
Comment (borkmann) [reply bpf/include/bpf/verifier.h]: Yep, makes sense, done.

**Comment by borkmann** [comment]
test-me-please

**Review by vadorovsky — APPROVED** [review]

**Pull request merged**